src/main/java/com/greenfox/tribes/controllers/ActivityController.java

```diff
@@ -107,6 +109,45 @@ public String pve(Model model) {
     Authentication auth = SecurityContextHolder.getContext().getAuthentication();
     WastelandUser user = userRepository.findByUsername(auth.getName()).get();
     Persona userHero = userRepository.findById(user.getPersona().getId()).get().getPersona();
+
+
+    ActivityDTO dto = activityService.getActivity(userHero.getId());
+    ActivityType type;
+    if (dto != null && dto.getType() != null) {
```

Comment:
Methods should be this long, especially in the controller. Consider refactoring this